File: frontend/src/schemas/cockpit.ts
/**
 * Zod schemas for HealthRule and Cockpit evaluation API.
 * MUST stay in sync with backend/recipe/schemas/cockpit.py
 */
import { z } from 'zod';

// ---------------------------------------------------------------------------
// HealthRule
// ---------------------------------------------------------------------------

export const HealthRuleSchema = z.object({
  id: z.number(),
  name: z.string(),
  description: z.string(),
  parameter: z.string(),
  scope: z.string(),
  threshold_green: z.number(),
  threshold_yellow: z.number(),
  unit: z.string(),
  tip_text: z.string(),
  is_active: z.boolean(),
  sort_order: z.number(),
});
export type HealthRule = z.infer<typeof HealthRuleSchema>;

// ---------------------------------------------------------------------------
// CockpitEvaluation (single rule result)
// ---------------------------------------------------------------------------

export const CockpitEvaluationSchema = z.object({
  rule_id: z.number(),
  rule_name: z.string(),
  parameter: z.string(),
  current_value: z.number(),
  status: z.string(), // "green" | "yellow" | "red"
  tip_text: z.string(),
  unit: z.string(),
});
export type CockpitEvaluation = z.infer<typeof CockpitEvaluationSchema>;

// ---------------------------------------------------------------------------
// CockpitDashboard (full response)
// ---------------------------------------------------------------------------

export const CockpitDashboardSchema = z.object({
  evaluations: z.array(CockpitEvaluationSchema),
  summary_status: z.string(), // worst status across all evaluations
  green_count: z.number(),
  yellow_count: z.number(),
  red_count: z.number(),
});
export type CockpitDashboard = z.infer<typeof CockpitDashboardSchema>;

// ---------------------------------------------------------------------------
// Traffic light colors (for UI)
// ---------------------------------------------------------------------------

export const COCKPIT_STATUS_COLORS: Record<string, { bg: string; text: string; label: string }> = {
  green: { bg: 'bg-green-500', text: 'text-white', label: 'Gut' },
  yellow: { bg: 'bg-yellow-400', text: 'text-black', label: 'Achtung' },
  red: { bg: 'bg-red-500', text: 'text-white', label: 'Kritisch' },
};

File: backend/recipe/schemas/cockpit.py
"""HealthRule and cockpit evaluation schemas."""

from ninja import Schema


class HealthRuleOut(Schema):
    """Schema for a single health rule."""

    id: int
    name: str
    description: str
    parameter: str
    scope: str
    min_green: float | None
    min_yellow: float | None
    max_green: float | None
    max_yellow: float | None
    unit: str
    tip_text: str
    is_active: bool
    sort_order: int


class HealthRuleIn(Schema):
    """Input schema for creating a health rule."""

    name: str
    description: str = ""
    parameter: str
    scope: str
    min_green: float | None = None
    min_yellow: float | None = None
    max_green: float | None = None
    max_yellow: float | None = None
    unit: str = ""
    tip_text: str = ""
    is_active: bool = True
    sort_order: int = 0


class HealthRuleUpdateIn(Schema):
    """Partial update schema for a health rule."""

    name: str | None = None
    description: str | None = None
    parameter: str | None = None
    scope: str | None = None
    min_green: float | None = None
    min_yellow: float | None = None
    max_green: float | None = None
    max_yellow: float | None = None
    unit: str | None = None
    tip_text: str | None = None
    is_active: bool | None = None
    sort_order: int | None = None


class CockpitEvaluationOut(Schema):
    """Schema for a single cockpit evaluation result."""

    rule_id: int
    rule_name: str
    parameter: str
    current_value: float
    status: str  # "green", "yellow", "red"
    tip_text: str
    unit: str


class CockpitDashboardOut(Schema):
    """Schema for a complete cockpit dashboard response."""

    evaluations: list[CockpitEvaluationOut]
    summary_status: str  # worst status across all evaluations
    green_count: int
    yellow_count: int
    red_count: int
